### PR TITLE
testing/libguestfs: new aport

### PR DESCRIPTION
**Pull request by fboudra**
Closes: https://bugs.alpinelinux.org/issues/1792

Signed-off-by: Fathi Boudra <fathi.boudra@linaro.org>

### CODE REVIEW

File: testing/libguestfs/APKBUILD [hunk 1]
```diff
@@ -0,0 +1,114 @@
+# Contributor:
+# Maintainer: Fathi Boudra <fathi.boudra@linaro.org>
+pkgname=libguestfs
+pkgver=1.40.2
+pkgrel=0
+pkgdesc="libguestfs"
```
Comment (Ikke) [inline testing/libguestfs/APKBUILD]: Could you improve this description? It should be more then just the package name repeated.

File: testing/libguestfs/APKBUILD [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+		--disable-lua \
+		--disable-golang \
+		--disable-gobject \
+		--without-java
+	# Building index-parse.c and index-scan.c by hand works around a race
+	# condition in the autotools cruft, where two other more copies of yacc
+	# race with each other, resulting in a corrupted file.
+	make -j1 -C builder index-parse.c
+	make -j1 -C builder index-scan.c
+	make -j$(nproc) V=1 INSTALLDIRS=vendor LTLIBINTL=-lintl
```
Comment (Ikke) [inline testing/libguestfs/APKBUILD]: abuild will use the amount of processes specified in /etc/abuild.conf, you should not specify it here (imagine someone not wanting to use all CPUs when building).

File: testing/libguestfs/APKBUILD [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+}
+
+check() {
+	cd "$builddir"
+	: make quickcheck
```
Comment (Ikke) [inline testing/libguestfs/APKBUILD]: This is a noop, so effectively no tests are run. Either run tests, or provide !check to options with a comment why checks are skipped.

**Comment by fboudra** [comment]
@Ikke Thanks for the review. All the comments have been addressed.

Wrt tests, I commented but kept check() for reference purpose if someone is interested to run them locally. I'd like to get it fixed by adding supermin dependency (https://github.com/fboudra/aports/commit/6f08b79c3e683823b0817388f13a913a68898a78), however it requires more work since the testsuite isn't distro-agnostic and Alpine isn't a platform officially supported by supermin.
Note: apk package manager support can certainly be contributed to supermin and we'll eventually be able to create Alpine based supermin appliance ;) Unfortunately, it's out of my current scope.

**Comment by fboudra** [comment]
FYI supermin submitted as https://github.com/alpinelinux/aports/pull/6505

**Pull request closed without merging**